lib/command.js

```diff
@@ -780,34 +794,61 @@ Expecting one of '${allowedValues.join("', '")}'`);
    */
 
   setOptionValue(key, value) {
-    return this.setOptionValueWithSource(key, value, undefined);
+    return this.setOptionValueWithSource(key, value);
   }
 
   /**
     * Store option value and where the value came from.
     *
     * @param {string} key
     * @param {Object} value
-    * @param {string} source - expected values are default/config/env/cli/implied
+    * @param {string} [source]
     * @return {Command} `this` command for chaining
     */
 
   setOptionValueWithSource(key, value, source) {
+    const set = this._asyncParsing === undefined
+      ? this._setPersistentOptionValueWithSource
+      : this._setNonPersistentOptionValueWithSource;
+    set(key, value, source);
```

Comment:
I suspect you may need to use "bind" or "apply" to the method to have `this` defined correctly inside the called routine?

(I hit an `undefined` error while manually testing and had this on call stack.)

Reply:
Steps to reproduce:

```
import { Command, Option } from 'commander';
const program = new Command();
program.setOptionValue('beforeFirstParse', "1");
```

Reply:
Resolved in https://github.com/tj/commander.js/pull/1919/commits/56eed63dee40a7000aac08fd08197c641c3f6e70 (I think?)